lib/Code/TidyAll/Git/Precommit.pm

```diff
@@ -41,8 +41,8 @@ sub check {
         # Store the stash, and restore it upon exiting this scope
         unless ( $self->no_stash ) {
             run( $self->git_path, "stash", "-q", "--keep-index" );
-            scope_guard { run( $self->git_path, "stash", "pop", "-q" ) };
         }
+        scope_guard { run( $self->git_path, "stash", "pop", "-q" ) unless $self->no_stash };
```

Comment:
I'm still not sure this is right. I think you need to write:

```perl
my $guard = scope_guard { ... };
```

Otherwise the guard would fire immediately, from my reading of the `Scope::Guard` code. There's really no way this could have ever worked.

I think it should be written as:

```perl
my $guard;
unless ( $self->no_stash ) {
    run( $self->git_path, "stash", "-q", "--keep-index" );
    $guard = scope_guard {
        run( $self->git_path, "stash", "pop", "-q" );
    };
}
```

It'd be really nice to have a test for this, especially since messing with someone's stash and accidentally popping _too much_ can cause a huge mess.